src/actions/market.ts

```diff
@@ -0,0 +1,85 @@
+  const abi = parseAbi([
+    'function getMarkets(address reader) view returns (((address token, uint8 decimals, string symbol) tkn0, (address token, uint8 decimals, string symbol) tkn1, uint256 tickSpacing)[])',
+  ])
+  const { mgvReader } = actionParams
+
+  const data = await getAction(
+    client,
+    call,
+    'call',
+  )({
+    code: bytecode,
+    data: encodeFunctionData({
+      abi: abi,
+      functionName: 'getMarkets',
+      args: [mgvReader],
+    }),
+  })
+  const markets = decodeFunctionResult({
+    abi: abi,
+    functionName: 'getMarkets',
+    data: data.data!,
+  })
```

Comment:
You can simplify this by using readContract instead: https://viem.sh/docs/contract/readContract#bytecode

```suggestion
  const markets = await getAction(
    client,
    readContract,
    'readContract',
  )({
    code: bytecode,
    abi: abi,
    functionName: 'getMarkets',
    args: [mgvReader]
  })
```